src/category_theory/limits/shapes/finite_limits.lean

```diff
@@ -31,41 +31,29 @@ This is often called 'finitely complete'.
 -/
 -- We can't just made this an `abbreviation`
 -- because of https://github.com/leanprover-community/lean/issues/429
-def has_finite_limits : Prop :=
-Π (J : Type v) [𝒥 : small_category J] [@fin_category J 𝒥], @has_limits_of_shape J 𝒥 C _
+class has_finite_limits : Prop :=
+(out (J : Type v) [𝒥 : small_category J] [@fin_category J 𝒥] : @has_limits_of_shape J 𝒥 C _)
```

Comment:
I'm not happy about these generic `out` names everywhere.  How about `to_has_limits_of_shape` or `limits_of_shape` or `of_shape` instead?

Reply:
I'm not sure what the point of a descriptive name here is. It may as well be a completely generic projection, and the name of the class is already in the projection: `has_finite_limits.out`. I would be using `h.1` but that doesn't always work because it takes `h` as a typeclass argument. Making this something giant like `h.to_has_limits_of_shape` makes the ergonomic loss much more significant.